src/socket_type.cpp

```diff
@@ -72,19 +72,19 @@ namespace libtorrent
 
 	bool is_utp(socket_type const& s)
 	{
-		return s.get<utp_stream>()
+		return (s.get<utp_stream>() != NULL)
```

Comment:
the parentheses here seem perhaps a bit excessive (and around the second term too)